### ISSUE TITLE
Move `queueAnimation` to component

### ISSUE THREAD

**Issue by grabbou** [Kureev/react-native-side-menu#74]
So it's overridable and we can easily ignore `animationKind` and provide own LayoutAnimations. Just an idea, as I am not that happy with the current one and I am investigating changing the LayoutAnimations to something different at all.

**Comment by Kureev** [comment]
I'd like to implement `react-motion` for that. Just waiting it to be a bit more stable and tested.

**Comment by grabbou** [comment]
Hm, what about `Animated`?

**Comment by Kureev** [comment]
@grabbou Animated?

**Comment by grabbou** [comment]
No, I am talking about `Animated` - https://facebook.github.io/react-native/docs/animations.html#animated they presented that library during ReactEurope and looks like a quite powerful tool to do native animations (it looks a lot like CoreAnimations in terms of how you group different bits, but haven't dig into the details yet) :dancer:

**Comment by Kureev** [comment]
Now I'm already using it and you can pass `animation` prop to component. Am I missing something? What's the idea you want me to implement?

**Comment by grabbou** [comment]
Nah you are using `LayoutAnimation`, just saying `Animated` is worth investigating as a good alternative to `react-motion`

**Comment by Kureev** [comment]
That's true. So, you want to animate opening/closing of the side menu. What is your propose for API? something like
```javascript
const queue = [
  Animated.decay(position, {   // coast to a stop
    velocity: {x: gestureState.vx, y: gestureState.vy}, // velocity from gesture release
    deceleration: 0.997,
  }),
  Animated.parallel([          // after decay, in parallel:
    Animated.spring(position, {
      toValue: {x: 0, y: 0}    // return to start
    }),
    Animated.timing(twirl, {   // and twirl
      toValue: 360,
    }),
  ]),
];
// ...
<SideMenu animationQueue={queue}/>
```
right?
And in the background it may be used like this:
```javascript
updatePosition() {
  // this.sideMenu.setNativeProps({ left: this.left, });
  Animated.queue(this.prop.animationQueue).start();
}
```
where `this.state.left = Animated.Value(0);`?

**Comment by Kureev** [comment]
Should be solved in `0.12.1`
